packages/blade/src/components/Tabs/TabIndicator.tsx

```diff
@@ -0,0 +1,139 @@
+  const { selectedValue, baseId, variant } = useTabsContext();
+  const [hasMeasured, setHasMeasured] = React.useState(false);
+  const [activeElementDimensions, setActiveElementDimensions] = React.useState({
+    width: 0,
+    height: 0,
+    x: 0,
+    y: 0,
+  });
+
+  // Update the dimensions of the active element
+  const updateDimensions = React.useCallback(() => {
+    const tabItemId = `${baseId}-${selectedValue}-tabitem`;
+    const activeTabItem = document.getElementById(tabItemId);
+    if (!activeTabItem) return;
+
+    setActiveElementDimensions({
+      width: activeTabItem.offsetWidth,
+      height: activeTabItem.offsetHeight,
+      x: activeTabItem.offsetLeft,
+      y:
+        variant === 'filled'
+          ? // on filled variant the indicator is positioned on top of the tab item
+            // so no need to add offsetHeight
+            activeTabItem.offsetTop - 1
+          : activeTabItem.offsetTop + activeTabItem.offsetHeight - 1.5,
+    });
+  }, [baseId, selectedValue, tabListContainerRef, variant]);
+
+  // Update the dimensions when the selected value changes
+  useIsomorphicLayoutEffect(() => {
+    if (!selectedValue) return;
+
+    updateDimensions();
+
+    const id = requestAnimationFrame(() => {
+      setHasMeasured(true);
+    });
+
+    return () => {
+      if (id) {
+        cancelAnimationFrame(id);
+      }
+    };
+  }, [baseId, selectedValue]);
+
+  // Update the dimensions when the window resizes or when the font loads
+  React.useEffect(() => {
+    if (!tabListContainerRef.current) return;
+
+    // check for FontFace API support
+    // FontFaceAPI is widely supported but better to be safe than sorry
+    if ('fonts' in document) {
+      try {
+        // wait for fonts to be loaded and then recalculate the dimensions
+        void document.fonts.ready.then(() => {
+          updateDimensions();
+        });
+      } catch (err: unknown) {
+        /* empty */
+      }
+    }
+
+    window.addEventListener('resize', updateDimensions);
+
+    return () => {
+      if (!tabListContainerRef.current) return;
+      window.removeEventListener('resize', updateDimensions);
+    };
+  }, [tabListContainerRef, updateDimensions]);
+
+  const transitionProps = {
+    transitionProperty: 'transform, width, background-color',
+    transitionDuration: hasMeasured
+      ? castWebType(makeMotionTime(theme.motion.duration.gentle))
+      : '0ms',
+    transitionTimingFunction: castWebType(theme.motion.easing.standard.effective),
+  };
+
+  if (variant === 'filled') {
+    return (
+      <StyledTabIndicator
+        position="absolute"
+        left="0px"
+        top="0px"
+        borderRadius="small"
+        backgroundColor="brand.primary.300"
+        style={{
+          ...transitionProps,
+          width: `${activeElementDimensions.width}px`,
+          height: `${activeElementDimensions.height}px`,
+          transform: `translate(${activeElementDimensions.x}px, ${activeElementDimensions.y}px)`,
```

Comment:
nit: makeSize?

Reply:
It's fine I think, this component is only for web. makeSize makes the LOC too long and hard to read.